guides/docker_guidance.md

```diff
@@ -0,0 +1,217 @@
+# Docker guidance
```

Comment:
As someone completely new to Docker, could we have a short sentence explaining what it does here, or a link to something that explains it?

Reply:
I've added a brief introduction to what containers are and how that relates to Docker with a link to the Docker docs to give some context.

Reply:
Brilliant, thanks 👍